wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java

```diff
@@ -478,11 +397,11 @@ private void getTypes(Type parent, List<Type> types) {
   }
 
   private void emitRegistry() throws IOException {
-    int packageClassSep = registryClass.lastIndexOf(".");
-    String javaPackage = registryClass.substring(0, packageClassSep);
-    String className = registryClass.substring(packageClassSep + 1);
+    int packageClassSep = options.registryClass.lastIndexOf(".");
```

Comment:
I missed this comment on my earlier pass.  Since parsing the class name didn't seem to have to do with command line parsing, I left this out for now.  I thought about moving all command line argument validation into the same place, but I couldn't find a great reason to move it all into CommandLineOptions.